### PR TITLE
fixed point vm_module tests

### PR DESCRIPTION
**Pull request by baykaner**

### CODE REVIEW

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 1]
```diff
@@ -28,21 +30,173 @@ class FixedPointTest : public ::testing::Test
   VmTestToolkit     toolkit{&stdout};
 };
 
-TEST_F(FixedPointTest, create_fixed_point)
+bool RunTest(VmTestToolkit &toolkit, std::stringstream &stdout, char const *TEXT, double gt)
 {
   auto m = toolkit.module();
+  EXPECT_TRUE(toolkit.Compile(TEXT));
+  EXPECT_TRUE(toolkit.Run());
+  EXPECT_EQ(std::stod(stdout.str()), gt);
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: Use EXPECT_DOUBLE_EQ to compare values of type `double`

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 2]
```diff
@@ -28,21 +30,173 @@ class FixedPointTest : public ::testing::Test
   VmTestToolkit     toolkit{&stdout};
 };
 
-TEST_F(FixedPointTest, create_fixed_point)
+bool RunTest(VmTestToolkit &toolkit, std::stringstream &stdout, char const *TEXT, double gt)
 {
   auto m = toolkit.module();
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: `m` appears unused

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 3]
```diff
@@ -28,21 +30,173 @@ class FixedPointTest : public ::testing::Test
...
+        myArray[i] = toFixed64(i);
+      endfor
+      print(myArray[3]);
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp64_t(3));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+//
+// TEST_F(FixedPointTest, map_32_fixed_point)
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: This is not good practice - commented out tests are easy to forget about. This test looks reasonable. If the underlying functionality is not implemented yet, uncomment the test and call it `DISABLED_map_32_fixed_point`, then gtest will ignore it and report that there's a disabled test.

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 4]
```diff
@@ -28,21 +30,173 @@ class FixedPointTest : public ::testing::Test
...
+    function main()
+      var pi : Fixed32 = 3.1415fp32;
+      print(cos(pi));
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::math::Cos(fetch::fixed_point::fp32_t(3.1415)));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+// TEST_F(FixedPointTest, exp_64_fixed_point)
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: uncomment, rename to `DISABLED_exp_64_fixed_point`

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 5]
```diff
@@ -28,21 +30,173 @@ class FixedPointTest : public ::testing::Test
...
+//    function main()
+//      var val : Fixed64 = 1fp64;
+//      print(exp(val));
+//    endfunction
+//  )";
+//  double gt = static_cast<double>(fetch::fixed_point::fp64_t(2.71828182846));
+//  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+//}
+//
+// TEST_F(FixedPointTest, pow_64_fixed_point)
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: `DISABLED_pow_64_fixed_point`

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 6]
```diff
@@ -28,21 +30,173 @@ class FixedPointTest : public ::testing::Test
...
+//  )";
+//  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1));
+//  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+//}
+
+TEST_F(FixedPointTest, sin_pi_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var pi : Fixed32 = 3.1415fp32;
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: No need to specify the type when you define a variable using a literal: `var pi = 3.1415fp32;`. Ditto other tests here

**Review by wkaluza — CHANGES_REQUESTED** [review]

**Review by markos — DISMISSED** [review]
LGTM

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 7]
```diff
@@ -28,21 +31,158 @@ class FixedPointTest : public ::testing::Test
...
-  ASSERT_TRUE(toolkit.Run());
+TEST_F(FixedPointTest, exp_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var val : Fixed32 = 1fp32;
+      print(exp(val));
+    endfunction
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(2.71828182846));
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: Why not `fp32_t::CONST_E`? Analogous to `fp32_t::CONST_PI` above.

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 8]
```diff
@@ -28,21 +31,158 @@ class FixedPointTest : public ::testing::Test
...
+  )";
+  double      gt   = static_cast<double>(fetch::fixed_point::fp32_t(1));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+TEST_F(FixedPointTest, sin_pi_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var pi : Fixed32 = 2fp32;
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: Redundant explicit type.

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 9]
```diff
@@ -28,21 +31,158 @@ class FixedPointTest : public ::testing::Test
...
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
 
-  ASSERT_TRUE(toolkit.Compile(TEXT));
-  ASSERT_TRUE(toolkit.Run());
+TEST_F(FixedPointTest, exp_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var val : Fixed32 = 1fp32;
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: Redundant explicit type.

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 10]
```diff
@@ -28,21 +31,158 @@ class FixedPointTest : public ::testing::Test
...
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
 
-  double gt = static_cast<double>(fetch::fixed_point::fp32_t(1));
-  EXPECT_EQ(std::stod(stdout.str()), gt);
+TEST_F(FixedPointTest, pow_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var val : Fixed32 = 2fp32;
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: Redundant explicit type.

File: libs/vm_modules/tests/unit/fixed_point.cpp [hunk 11]
```diff
@@ -28,21 +31,158 @@ class FixedPointTest : public ::testing::Test
...
+  )";
+  double      gt   = static_cast<double>(fetch::math::Sin(fetch::fixed_point::fp32_t(2)));
+  EXPECT_TRUE(RunTest(toolkit, stdout, TEXT, gt));
+}
+
+TEST_F(FixedPointTest, cos_pi_32_fixed_point)
+{
+  char const *TEXT = R"(
+    function main()
+      var pi : Fixed32 = 3.1415fp32;
```
Comment (wkaluza) [inline libs/vm_modules/tests/unit/fixed_point.cpp]: Redundant explicit type.

**Review by wkaluza — CHANGES_REQUESTED** [review]

**Review by wkaluza — DISMISSED** [review]

**Review by wkaluza — APPROVED** [review]

**Pull request merged**